File: lib/src/features/admin/management/domain/race.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'package:hero/src/features/admin/management/domain/attribute_value.dart';

part 'race.freezed.dart';
part 'race.g.dart';

@freezed
class Race with _$Race {
  const factory Race({
    required String id,
    required String name,
    String? description,
    @Default([]) List<AttributeValue> attributes,
  }) = _Race;

  factory Race.fromJson(Map<String, dynamic> json) => _$RaceFromJson(json);
}

File: Server/Hero.Server.Core/Models/Race.cs
using System;

namespace Hero.Server.Core.Models
{
    public class Race
    {
        public Guid Id  { get; set; }
        public Guid GroupId { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }

        public List<AttributeRace> Attributes { get; set; }

        public void Update(Race race)
        {
            this.Name = race.Name;
            this.Description= race.Description;
            this.Attributes = race.Attributes;
        }
    }
}
